### PR TITLE
fix compiler failure with regexp_replace function when working with Hive tables containing row(map<varchar, varchar>) data type

### LINKED ISSUE #19689: Problem with regexp_replace function when working with Hive tables containing row(map<varchar, varchar>) data type
I encountered an issue when using the regexp_replace function in Trino with Hive tables that have a field of type row(map<varchar, varchar>). The problem is that when attempting to search and replace values within the map, I am experiencing a bug.  

Please refer to the attached screenshot, which illustrates the problem.

企业微信截图_e54c067d-f620-4cf2-a8bf-6830de652efb

### Steps to reproduce:
1. Create a Hive table with a column of type row(map<varchar, varchar>).  e.g., `CREATE TABLE hive.tmp.trino_test_11_09 ( event ROW( event_info map(varchar, varchar) ) ) WITH (format = 'ORC');`
3. Execute a query using the regexp_replace function on that column, e.g., regexp_replace(map_column['key'], 'old_value', 'new_value'). e.g., `SELECT "regexp_replace"("event"."event_info" [ 'tags' ], '', '' ) FROM tmp.trino_test_11_09;`

### We are using Trino version 432

### PR DESCRIPTION
**Pull request by needlssj**
## Additional context and related issues
fix #19689

### CODE REVIEW

**Comment by needlssj** [comment]
Cla pdf on appendix





------------------ Original ------------------
From: cla-bot[bot] ***@***.***&gt;
Date: Wed,Nov 15,2023 4:06 PM
To: trinodb/trino ***@***.***&gt;
Cc: needlssj ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [trinodb/trino] fix compiler failure with regexp_replace functionwhen working with Hive tables containing row(map<varchar, varchar&gt;) data type(PR #19749)





 
Thank you for your pull request and welcome to the Trino community. We require contributors to sign our Contributor License Agreement, and we don't seem to have you on file. Continue to work with us on the review and improvements in this PR, and submit the signed CLA to ***@***.*** Photos, scans, or digitally-signed PDF files are all suitable. Processing may take a few days. The CLA needs to be on file before we merge your changes. For more information, see https://github.com/trinodb/cla

**Comment by mosabua** [comment]
:wave: @needlssj - this PR has become inactive. If you're still interested in working on it, please let us know, and we can try to get reviewers to help with that.

We're working on closing out old and inactive PRs, so if you're too busy or this has too many merge conflicts to be worth picking back up, we'll be making another pass to close it out in a few weeks.

**Comment by mosabua** [comment]
@findepi @findinpath @alexjo2144 @electrum is there merit in this PR and fix or should we close?

**Comment by electrum** [comment]
Apologies for the delay in reviewing this. Can you share a reproduction?

File: core/trino-main/src/main/java/io/trino/sql/planner/ConnectorExpressionTranslator.java [hunk 1]
```diff
@@ -807,8 +806,7 @@ private ConnectorExpression constantFor(Expression node)
                     ImmutableMap.of());
 
             if (type == JONI_REGEXP) {
-                Slice pattern = ((JoniRegexp) value).pattern();
-                return new Constant(pattern, createVarcharType(countCodePoints(pattern)));
```
Comment (findepi) [inline core/trino-main/src/main/java/io/trino/sql/planner/ConnectorExpressionTranslator.java:811]: this was here intetnionally. the JONI_REGEXP is considered engine's implementation detail and therefore we wanted connectors not to know about it

**Comment by takezoe** [comment]
This might have been solved by https://github.com/trinodb/trino/pull/20226.

**Comment by hashhar** [comment]
@takezoe indeed, thanks.

**Pull request closed without merging**